openssl.yaml

```diff
@@ -15,6 +15,12 @@ package:
   dependencies:
     runtime:
       - openssl-provider-legacy
+      # Melange will automatically add a dependency on 'so:libcrypto.so.3' and 'so:libssl.so.3' via SCA
+      # However, the version of OpenSSL must match exactly or the CLI will crash due to missing symbols
+      # Example: '/usr/lib/libcrypto.so.3: version `OPENSSL_3.4.0' not found (required by openssl)'
+      # To avoid this, we explicitly pin the versions of libcrypto3 and libssl3 to match the CLI version
+      - libcrypto3=${{package.full-version}}
+      - libssl3=${{package.full-version}}
 
```

Comment:
strict pin is not needed, as ">=" should be sufficient. I have pondered of doing this, but it will do very little.

Note all containers built prior to 3.4.0-r0 landing, have libcrypto3 & libssl3 pinned to ~3.3 version. And whilst 3.4.0-r2 will be excluded, instead 3.4.0-r1 will be installed. Unless we somehow rebuild all past openssl versions; or at least withdraw 3.4.0-r0 and 3.4.0-r1 (which breaks image rebuild reproducibility).

I'm happy to take this change, but it will likely only have meaningful effect with next upgrade to 3.5 in 6 months time. To avoid this same "experience" then.

Reply:
> strict pin is not needed, as ">=" should be sufficient. I have pondered of doing this, but it will do very little.

Would you prefer pinning to the same `<major>.<minor>.<patch>` version which is _technically_ all that's required for it to work:
```
      - libcrypto3>=${{package.version}}
```
Or instead the full-version including epoch:
```
      - libcrypto3>=${{package.full-version}}
```

Reply:
> I'm happy to take this change, but it will likely only have meaningful effect with next upgrade to 3.5 in 6 months time. To avoid this same "experience" then.

That's fine, we've worked around the issue by strictly pinning the installed version of `openssl` in the `apk add` command to match, this PR was mostly to avoid anyone having the same experience in the future if possible

Reply:
This by-hand change will fix installing openssl tool; which is a common operation but it will not fix this issue for all other packages. See for example "apk add python does not work" https://github.com/wolfi-dev/os/issues/33218 which is exactly this same issue.

I wish for melange to automatically generate versioned `so:LIB` provides, such that one can do depends `so:LIB>=$libversion` automatically. i opened https://github.com/chainguard-dev/melange/issues/1621 to track suck feature.

Please note this experience is not present with images built with apko; which always build images with a consistent set of packages; including self-service way of doing that for Chainguard Images customers.